packages/wp-story-editor/src/components/documentPane/status/status.js

```diff
@@ -78,19 +80,25 @@ function StatusPanel({ nameOverride }) {
 
   const isUploading = useIsUploadingToStory();
 
-  const [hasPassword, setHasPassword] = useState(Boolean(password));
-
-  const visibility = useMemo(() => {
-    if (hasPassword) {
-      return 'protected';
+  useEffect(() => {
+    if (password) {
+      updateStory({
+        properties: { visibility: 'protected' },
+      });
+      return;
     }
 
     if (status === 'private') {
-      return 'private';
+      updateStory({
+        properties: { visibility: 'private' },
+      });
+      return;
     }
 
-    return 'public';
-  }, [status, hasPassword]);
+    updateStory({
+      properties: { visibility: 'public' },
+    });
+  }, [password, status, updateStory]);
 
```

Comment:
Hi @swissspidy From what I gathered, we were setting `visibility` based on the story's status and whether or not a password was present. Basically, if a status was anything but `private` then it's `visibility` would be set to `public` _unless_ there was a `password` present. Since we now will be using this component in multiple places, I have added `visibility` to the story's state in order to keep the two instances of the component in sync. I am still setting `visibility` based on the presence of a `password`, so I wanted to make sure I was understanding that bit of logic correctly.

Reply:
Thanks for flagging! Yep, that's how it basically works in WP. So from what I can see this looks good 👍